src/groupadd.c

```diff
@@ -196,15 +198,15 @@ static void grp_update (void)
 #endif				/* SHADOWGRP */
 
 	if (user_list) {
-		char *token;
-		token = strtok(user_list, ",");
-		while (token) {
-			if (prefix_getpwnam (token) == NULL) {
-				fprintf (stderr, _("Invalid member username %s\n"), token);
+		char  *u, *ul;
```

Comment:
The block is:

```c
	if (user_list) {
		char  *u, *ul;

		ul = user_list;
		while (NULL != (u = strsep(&ul, ","))) {
			if (prefix_getpwnam(u) == NULL) {
				fprintf(stderr, _("Invalid member username %s\n"), u);
				exit (E_GRP_UPDATE);
			}
			grp.gr_mem = add_list(grp.gr_mem, u);
		}
	}
```

which has 1 line of declarations and 8 lines of code.  I think it makes sense to have short names for such temporary variables.  The first line of code already tells what the 'u' and the 'l' mean, so it'll be hard to be confused.  And having shorter names for short-lived variables makes it easier to distinguish in the head that something is short-lived.

I prefer the short names.

Reply:
<https://www.kernel.org/doc/html/v4.10/process/coding-style.html#naming>

Reply:
Also some good advice from Google's C++ style guide:

> Generally speaking, descriptiveness should be proportional to the name's scope of visibility. For example, n may be a fine name within a 5-line function, but within the scope of a class, it's likely too vague.

<https://google.github.io/styleguide/cppguide.html#General_Naming_Rules>

(Not saying that style guide says much good for C, but there are a few pieces of good advice.)

Reply:
I think we have a fair explanation of `ul` in this line (the first use of the variable):

```c
		char  *u, *ul;

		ul = user_list;
```

I think it's quite obvious.

The next line explains what `u` is, for which you need to at least have a rough idea of what strsep(3) does:

```c

		char  *u, *ul;

		ul = user_list;
		while (NULL != (u = strsep(&ul, ","))) {
```

`u` is an entry in the user list, so it can only be, of course, a user.